test/powershell/Language/Scripting/HashtableToPSCustomObjectConversion.Tests.ps1

```diff
@@ -0,0 +1,248 @@
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $x = [SampleClass2]@{blah=10;b=5 }
+                 Throw "Exception expected, execution should not have reached here"
+       }
+       catch
+       {
+           $_.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should Be 'ObjectCreationError'
+       }       
+    }
+
+
+    It 'Creating an object of an existing type from hashtable should throw error when setting incompatible values for properties' {       
+       
+       try
+       { 
+          $source = @"
+                        public class SampleClass3
+                        {
+                            public SampleClass3(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass3()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $x = [SampleClass3]@{a="foo";b=5}
+                 Throw "Exception expected, execution should not have reached here"
+       }
+       catch
+       {           
+           $_.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should Be 'ObjectCreationError'
+       }
+    }
+
+    #known issue 2069 
+    It 'Creating an object of an existing type from hashtable should call the constructor taking a hashtable if such a constructor exists in the type' -skip:$IsCoreCLR {
+       {
+                $source = @"
+                        public class SampleClass5
+                        {                                                       
+                            public SampleClass5(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass5(System.Collections.Hashtable h)   
+                            {
+		                          a = 100;
+		                          b = 200;
+                            }
+                            
+                            public SampleClass5()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $script:x = [SampleClass5]@{a=10;b=5}
+            
+       } | Should Not Throw
+       
+       $script:x | Should Not Be $null
+       ($script:x.a) | Should Be '100'
+    }

```

Comment:
I think we can use our own classes (it should be a valid scenario) (I'd like to avoid using add-type where we can as it's slow) - I think this breaks down to:
```
    It 'Creating an object of an existing type from hashtable should call the constructor taking a hashtable if such a constructor exists in the type' {
        class sample5 {
            [int]$a
            [int]$b
            Sample5([int]$x) { $this.a = $x }
            Sample5([hashtable]$h) { $this.a = 100; $this.b = 100 }
        }

        $x = [sample5]@{a=10;b=5}
        $x.a | should be 100
    }
```
And it should run on core, too!